docs/tutorials/custom_contracts/tezos.md

```diff
@@ -0,0 +1,626 @@
+
+> **NOTE:** For this tutorial we are going to use [SmartPy](https://smartpy.io/) for building Tezos smart contracts utilizing the broadly adopted Python language.
+
+## Example smart contract
+
+First let's look at a simple contract smart contract called `SimpleStorage`, which we will be using on a Tezos blockchain. Here we have one state variable called 'storedValue' and initialized with the value 12. During initialization the type of the variable was defined as 'int'. You can see more at [SmartPy types](https://smartpy.io/manual/syntax/integers-and-mutez). And then we added a simple test, which set the storage value to 15 and checks that the value was changed as expected.
+
+> **NOTE:** Tests are used to verify the validity of contract entrypoints and do not affect the state of the contract during deployment.
+
+Here is the source for this contract:
+
+```smarty
+import smartpy as sp
+
+@sp.module
+def main():
+    class SimpleStorage(sp.Contract):
+        def __init__(self, value):
+            self.data.storedValue = value
+
+        @sp.entrypoint
+        def replace(self, params):
+            self.data.storedValue = params.value
+
+@sp.add_test(name="SimpleStorage")
+def test():
+    c1 = main.SimpleStorage(12)
+    scenario = sp.test_scenario(main)
+    scenario.h1("SimpleStorage")
+    scenario += c1
+    c1.replace(value=15)
+    scenario.verify(c1.data.storedValue == 15)
+```
+
+## Contract deployment
+
+To deploy the contract, we will use [SmarPy IDE](https://smartpy.io/ide).
+1. Open an IDE;
+2. Paste the contract code;
+3. Click "Run code" button;
+4. Then you will see "Deploy Michelson Contract" button, click on that;
+5. Choose the Ghostnet network;
+6. Select an account, which you're going to use to deploy the contract;
+7. Click "Estimate Cost From RPC" button;
+8. Click "Deploy Contract" button;
+
+![ContractDeployment](images/tezos_contract_deployment.png)
+![ContractDeployment2](images/tezos_contract_deployment2.png)
+
+Here we can see that our new contract address is `KT1D254HTPKq5GZNVcF73XBinG9BLybHqu8s`. This is the address that we will reference in the rest of this guide.
+
+## The FireFly Interface Format
+
+As we know from the previous section - smart contracts on the Tezos blockchain are using the domain-specific, stack-based programming language called [Michelton](https://tezos.gitlab.io/active/michelson.html). It is a key component of the Tezos platform and plays a fundamental role in defining the behavior of smart contracts and facilitating their execution.
+This language is very efficient but also a bit tricky and challenging for learning, so in order to teach FireFly how to interact with the smart contract, we will be using [FireFly Interface (FFI)](../../reference/firefly_interface_format.md) to define the contract inteface which later will be encoded to Michelton.
+
+The following FFI sample demonstrates the specification for the widely used FA2 (analogue of ERC721 for EVM) smart contract:
```

Comment:
FFI for tezos smart contracts is not generated and it needs to be build manually, as well as for Fabric chaincode.
I used the example of the FA2 contract, since it covers most of the different data types and it is better to rely on it when building FFI for your own smart contract.
Later I plan to add an additional doc section for building FFI for tezos smart contracts with a review of all supported types and examples.